server/src/main/java/io/crate/expression/scalar/postgres/PgTableIsVisibleFunction.java

```diff
@@ -0,0 +1,112 @@
+ */
+
+package io.crate.expression.scalar.postgres;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+import io.crate.data.Input;
+import io.crate.exceptions.RelationUnknown;
+import io.crate.exceptions.SchemaUnknownException;
+import io.crate.expression.scalar.HasSchemaPrivilegeFunction;
+import io.crate.metadata.FunctionName;
+import io.crate.metadata.Functions;
+import io.crate.metadata.NodeContext;
+import io.crate.metadata.RelationInfo;
+import io.crate.metadata.RelationName;
+import io.crate.metadata.Scalar;
+import io.crate.metadata.Schemas;
+import io.crate.metadata.SearchPath;
+import io.crate.metadata.TransactionContext;
+import io.crate.metadata.functions.BoundSignature;
+import io.crate.metadata.functions.Signature;
+import io.crate.metadata.pgcatalog.OidHash;
+import io.crate.metadata.pgcatalog.PgCatalogSchemaInfo;
+import io.crate.metadata.settings.SessionSettings;
+import io.crate.metadata.table.Operation;
+import io.crate.role.Permission;
+import io.crate.role.Role;
+import io.crate.role.Roles;
+import io.crate.sql.tree.QualifiedName;
+import io.crate.types.DataTypes;
+
+public class PgTableIsVisibleFunction extends Scalar<Boolean, Integer> {
+
+    private static final FunctionName NAME = new FunctionName(PgCatalogSchemaInfo.NAME, "pg_table_is_visible");
+
+    public static void register(Functions.Builder module) {
+        module.add(
+            Signature.scalar(
+                NAME,
+                DataTypes.INTEGER.getTypeSignature(),
+                DataTypes.BOOLEAN.getTypeSignature()
+            ).withFeature(Feature.NULLABLE),
+            PgTableIsVisibleFunction::new);
+    }
+
+    protected PgTableIsVisibleFunction(Signature signature, BoundSignature boundSignature) {
+        super(signature, boundSignature);
+    }
+
+    @SafeVarargs
+    @Override
+    public final Boolean evaluate(TransactionContext txnCtx, NodeContext nodeContext, Input<Integer>... args) {
+        assert args.length == 1 : NAME + " expects exactly 1 argument, got " + args.length;
+        Integer tableOid = args[0].value();
+        if (tableOid == null) {
+            return null;
+        }
+
+        Schemas schemas = nodeContext.schemas();
+        RelationName relationNameFromOid = schemas.getRelation(tableOid);
+        if (relationNameFromOid == null) {
+            return false;
+        }
+
+        SessionSettings sessionSettings = txnCtx.sessionSettings();
+        Roles roles = nodeContext.roles();
+        Role user = roles.findUser(sessionSettings.userName());
+        List<String> searchPathsWithDQLPrivileges = new ArrayList<>();
+        for (String searchPath : sessionSettings.searchPath()) {
+            if (HasSchemaPrivilegeFunction.checkBySchemaOid(
+                roles,
+                user,
+                OidHash.schemaOid(searchPath),
+                List.of(Permission.DQL),
+                schemas)) {
```

Comment:
Looks like Postgres' privilege is a bit different:
```
-- 'usage' on schema does not imply the tables within 
=> select * from s2.t2;
ERROR:  permission denied for table t2
-- but still can perform `pg_table_is_visible`
=> select pg_table_is_visible('s2.t2'::regclass::oid);
 pg_table_is_visible 
---------------------
 t
(1 row)

```
And checking for dql on schema aligned with Postgres' behaviour but not consistent with CrateDB. I think we should check for table level privilege instead for CrateDB.